CMakeLists.txt

```
@@ -220,6 +220,28 @@ install(
   COMPONENT dev)
 
 find_package(CycloneDDS REQUIRED)
+# Support for shared memory in Cyclone DDS C++ depends on support for shared
+# memory being compiled into Cyclone DDS and iceoryx_binding_c being available
+# If Cyclone DDS is compiled with support for shared memory, Cyclone DDS C++
+# must be compiled with shared memory too and vice versa.
+get_target_property(cyclonedds_has_shm CycloneDDS::ddsc SHM_SUPPORT_IS_AVAILABLE)
+mark_as_advanced(cyclonedds_has_shm)
+
+option(ENABLE_SHM "Enable shared memory support" ${cyclonedds_has_shm})
+if(ENABLE_SHM)
+  if(NOT cyclonedds_has_shm)
+    message(FATAL_ERROR "Cyclone DDS is NOT compiled with support for shared memory")
+  else()
+    find_package(iceoryx_binding_c REQUIRED)
+  endif()
+  message(STATUS "Found iceoryx_binding_c: (found version \"${iceoryx_binding_c_VERSION}\")")
+  set(ENABLE_SHM ON)
```

```suggestion
  set(ENABLE_SHM TRUE CACHE BOOL "Iceoryx found!" FORCE)
```

This setting needs to be forced, otherwise it will not evaluated to true in the `PackageConfig.cmake.in`

---

I think this is working as expected also without forcing the option. If this was not evaluated correctly then the CI should fail when installing the HelloWorld example with SHM enabled, so probably I will leave this as it is for now